### PR TITLE
RAB-1280: Update unitary attribute group deletion modal

### PR DESCRIPTION
**Pull request by pierrejolly**
In this PR I updated the delete modal to allow user to move attributes to another attribute group

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/attribute-group/form/delete.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import {DeleteAttributeGroupModal} from '@akeneo-pim-community/settings-ui';
+
+class Delete extends ReactView {
+  reactElementToMount() {
+    const data = this.getRoot().getFormData();
+
+    return (
+      <DependenciesProvider>
+        <ThemeProvider theme={pimTheme}>
+          <DeleteAttributeGroupModal attributeGroupCode={data.code} />
```
Comment (ValentinMumble) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/attribute-group/form/delete.tsx:15]:
```suggestion
    const {code} = this.getRoot().getFormData();

    return (
      <DependenciesProvider>
        <ThemeProvider theme={pimTheme}>
          <DeleteAttributeGroupModal attributeGroupCode={code} />
```

File: src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/hooks/attribute-groups/useDeleteAttributeGroup.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  const deleteAttributeGroup = async (attributeGroupCode: string, replacementAttributeGroupCode: string | null) => {
+    startLoading();
+    const response = await fetch(router.generate(ATTRIBUTE_GROUP_DELETE_ROUTE, {identifier: attributeGroupCode}), {
+      method: 'DELETE',
+      headers: {
+        'Content-Type': 'application/json',
+        'X-Requested-With': 'XMLHttpRequest',
+      },
+      body: JSON.stringify({
+        replacement_attribute_group: replacementAttributeGroupCode,
```
Comment (ValentinMumble) [inline src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/hooks/attribute-groups/useDeleteAttributeGroup.ts:19]:
```suggestion
        replacement_attribute_group_code: replacementAttributeGroupCode,
```

File: src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/components/DeleteAttributeGroupModal.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  const [isDeleteModalOpen, openDeleteModal, closeDeleteModal] = useBooleanState(false);
+  const [replacementAttributeGroupCode, setReplacementAttributeGroupCode] = useState<string | null>(null);
+  const catalogLocale = useUserContext().get('catalogLocale');
+
+  const [isLoading, deleteAttributeGroup] = useDeleteAttributeGroup();
+  const [attributeGroups] = useAttributeGroups();
+  const attributeGroup = attributeGroups.find(attributeGroup => attributeGroup.code === attributeGroupCode);
+  if (!attributeGroup) return null;
+
+  const availableReplacementAttributeGroups = attributeGroups.filter(
```
Comment (ValentinMumble) [inline src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/components/DeleteAttributeGroupModal.tsx:32]: We may reuse this helper function to retrieve the list of impacted related attributes & replacement attribute groups
https://github.com/akeneo/pim-enterprise-dev/blob/23201107f8ff7269011a1f87f3f54134802d6391/src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/models/AttributeGroup.ts#L27

File: src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/components/DeleteAttributeGroupModal.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+                  'pim_enrich.entity.attribute_group.delete.select_attribute_group',
+                  {
+                    number_of_attribute: attributeGroup.attribute_count,
+                  },
+                  attributeGroup.attribute_count
+                )}
+              >
+                <SelectInput
+                  emptyResultLabel={translate('pim_enrich.entity.attribute_group.delete.empty_result_label')}
+                  onChange={(value: string | null) => setReplacementAttributeGroupCode(value)}
```
Comment (ValentinMumble) [inline src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/components/DeleteAttributeGroupModal.tsx:87]:
```suggestion
                  onChange={setReplacementAttributeGroupCode}
```

File: src/Akeneo/Pim/Structure/Bundle/Resources/translations/jsmessages.en_US.yml [hunk 5]
```diff
@@ -300,6 +300,13 @@ pim_enrich.entity.attribute_group:
         empty_result_label: 'No attribute group found'
         placeholder: 'Select attribute group'
         open_label: 'attribute group list'
+    delete:
+        confirm: 'Are you sure you want to delete this attribute group'
+        attribute_warning: 'Deleting this attribute group will impact {{ number_of_attribute }} child attribute. You will need to move this child attribute to a new attribute group.'
+        select_attribute_group: '{1}Please select the attribute group to move the child attribute to:|]1, Inf[Please select the attribute group to move the {{ number_of_attribute }} child attributes to:'
+        empty_result_label: 'No attribute group found'
+        placeholder: 'Select attribute group'
+        open_label: 'attribute group list'
```
Comment (ValentinMumble) [inline src/Akeneo/Pim/Structure/Bundle/Resources/translations/jsmessages.en_US.yml:309]: These translations can be reused from the mass delete IMO, as well as the `pim_enrich.entity.attribute_group.delete.confirm`

**Review by ValentinMumble — APPROVED** [review]

**Review by cemorin — APPROVED** [review]

**Review by MadeWilson — APPROVED** [review]

**Pull request merged**